src/test/java/com/puppycrawl/tools/checkstyle/utils/TokenUtilTest.java

```diff
@@ -395,4 +395,34 @@ public void testIsBooleanLiteralType() {
                 .isFalse();
     }
 
+    @Test
+    public void testIdentIsUnnamedTrue() {
+        final DetailAstImpl identifierAst = new DetailAstImpl();
+        identifierAst.setType(TokenTypes.IDENT);
+        identifierAst.setText("_");
+        final boolean result = TokenUtil.isUnnamed(identifierAst);
+
+        assertWithMessage("Result is not expected")
```

Comment:
All tests should be through real check execution.
In other cases we can generate fancy utils with fancy code and and fancy pure junit test method and will still not work in real code and real Check.

Reply:
We already had bunch of such examples in past.
Pure unit testing is forbidden, and allowed only with huge comment above to explain all nuances of why it is not possible to do it through Check. 
Unfortunately there are bunch bad examples in our code base, we were not smart from the beginning.

Reply:
Yes, this method will be tested using real check execution once we support unnamed variables in the checks. I think this is the same for all utils here. all TokenUtil Tests are pure JUnit tests but at the same time they are being tested using real check execution in the checks test

Reply:
So, I think pureJunit here in TokenUtil is ok as long as we use these utils in some checks then we are sure that they are tested in real check execution in the checks test

Reply:
I have a claim it may be wrong but we can even remove almost all tests from `TokenUtilTest` and code coverage and pitest will not complain if we use the utils method appropriately in the checks

Reply:
I would like to remove all and make sure all is covered by regular Check tests, one day we will do this.